translate/storage/pypo.py

```diff
@@ -759,16 +761,16 @@ class pofile(pocommon.pofile):
 
     def parse(self, input):
         """Parses the given file or file source string."""
-        if True:
-            if hasattr(input, 'name'):
-                self.filename = input.name
-            elif not getattr(self, 'filename', ''):
-                self.filename = ''
-            if isinstance(input, str):
-                input = BytesIO(input)
-            # clear units to get rid of automatically generated headers before parsing
-            self.units = []
-            poparser.parse_units(poparser.ParseState(input, pounit), self)
+        if hasattr(input, 'name'):
+            self.filename = input.name
+        elif not getattr(self, 'filename', ''):
+            self.filename = ''
+        if isinstance(input, str):
+            input = BytesIO(input)
+        # clear units to get rid of automatically generated headers before parsing
+        self.units = []
+        poparser.parse_units(poparser.ParseState(input, pounit), self)
+        self.removeduplicates(duplicatestyle="discard")
```

Comment:
Please add a commit that cleans that superfluous `if` so we can separate the cleanup for the actual changes.